### PR TITLE
feat(validation): added validation error handling

### LINKED ISSUE #20: Check for openapi validation error in global error handler

### PR DESCRIPTION
**Pull request by cl-shaharshavit-rnd**
closes #20

### CODE REVIEW

**Review by asafMasa — APPROVED** [review]

File: projectTemplates/ts-server/src/middleware/ErrorHandler.ts [hunk 1]
```diff
@@ -15,10 +17,14 @@ export class ErrorHandler {
...
-      );
-      res.sendStatus(StatusCodes.INTERNAL_SERVER_ERROR);
+      if (err instanceof InputValidationError) {
+        res.status(HttpStatus.BAD_REQUEST).json(err.errors);
+      } else {
+        this.logger.error(
+          `${req.method} request to ${req.originalUrl}  has failed with error: ${err.message}`
+        );
+        res.sendStatus(StatusCodes.INTERNAL_SERVER_ERROR);
+      }
```
Comment (syncush) [inline projectTemplates/ts-server/src/middleware/ErrorHandler.ts]: depends on NODE_ENV 

if NODE_ENV is production then we want to return status of 500 
but if it NODE_DEV=development we would like to output thee message and stacktrace

File: projectTemplates/ts-server/src/middleware/ErrorHandler.ts [hunk 2]
```diff
@@ -2,6 +2,8 @@ import { MCLogger } from '@map-colonies/mc-logger';
 import { Request, Response, NextFunction, ErrorRequestHandler } from 'express';
 import { injectable } from 'tsyringe';
 import { StatusCodes } from 'http-status-codes';
+import { InputValidationError } from 'openapi-validator-middleware';
+import HttpStatus from 'http-status-codes';
```
Comment (syncush) [inline projectTemplates/ts-server/src/middleware/ErrorHandler.ts]: you import `http-status-codes` twice

**Review by syncush — CHANGES_REQUESTED** [review]

**Review by asafMasa — APPROVED** [review]

**Review by syncush — APPROVED** [review]

File: projectTemplates/ts-server/src/middleware/ErrorHandler.ts (hunk 1 above)
Comment (cl-shaharshavit-rnd) [reply projectTemplates/ts-server/src/middleware/ErrorHandler.ts]: by default the NODE_ENV seams empty.
do we want to add it to .env.example?
or do we want to check if it is not "production"? seams more risky but is set in dockerfile

File: projectTemplates/ts-server/src/middleware/ErrorHandler.ts (hunk 2 above)
Comment (cl-shaharshavit-rnd) [reply projectTemplates/ts-server/src/middleware/ErrorHandler.ts]: fixed

**Review by asafMasa — APPROVED** [review]

**Review by syncush — APPROVED** [review]

**Pull request merged**